packages/@aws-cdk/aws-ecs-patterns/lib/base/application-load-balanced-service-base.ts

```diff
@@ -313,21 +322,25 @@ export abstract class ApplicationLoadBalancedServiceBase extends cdk.Construct {
     if (props.certificate !== undefined && props.protocol !== undefined && props.protocol !== ApplicationProtocol.HTTPS) {
       throw new Error('The HTTPS protocol must be used when a certificate is given');
     }
+
     const protocol = props.protocol !== undefined ? props.protocol :
       (props.certificate ? ApplicationProtocol.HTTPS : ApplicationProtocol.HTTP);
 
+    const containerProtocol = props.containerProtocol !== undefined ? props.containerProtocol : ApplicationProtocol.HTTP;
+
     const targetProps = {
-      port: 80
+      port: 80,
```

Comment:
Now that I think of it would we also need to be able to override the container port?